addons/hr_expense/tests/test_expenses.py

```diff
@@ -585,26 +588,242 @@ def test_expense_corner_case_changing_employee(self):
                 'price_unit': 1000.00,
-                'date': '2020-01-02',
+                'date': '2022-01-01',
+            })],
+        })
+
+        expense_sheet.action_submit_sheet()
+
+        with freeze_time(self.frozen_today):
+            expense_sheet.action_approve_expense_sheets()
+            expense_sheet.action_sheet_move_create()
+
+        self.assertEqual(expense_sheet.accounting_date, date(year=2022, month=1, day=25))
+
+    def test_computation_expense_report_date_with_most_recent_expense_within_month_later(self):
+        """
+            Test the accounting date if the most recent expense is within this month but after today
+            The accounting date should then be today
+        """
+        expense_sheet_2 = self.create_expense_report({
+            'name': 'Expense for John Smith',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1000.00,
+                'date': '2022-01-29',
+            })],
+        })
+        expense_sheet_2.action_submit_sheet()
+
+        with freeze_time(self.frozen_today):
+            expense_sheet_2.action_approve_expense_sheets()
+            expense_sheet_2.action_sheet_move_create()
+
+        self.assertEqual(expense_sheet_2.accounting_date, date(year=2022, month=1, day=25))
+
+    def test_computation_expense_report_date_with_most_recent_expense_last_month(self):
+        """
+            Test the accounting date if the most recent expense is before this month and there is no lock date
+            The accounting date should then be the last day of the mst recent expense month
+        """
+        expense_sheet = self.create_expense_report({
+            'name': 'Expense for John Smith',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1000.00,
+                'date': '2021-12-20',
+            })],
+        })
+        expense_sheet.action_submit_sheet()
+
+        with freeze_time(self.frozen_today):
+            expense_sheet.action_approve_expense_sheets()
+            expense_sheet.action_sheet_move_create()
+
+        # no lock date so defaults to last day of month of the most recent expense
+        self.assertEqual(expense_sheet.accounting_date, date(year=2021, month=12, day=31))
+
+        expense_sheet_2 = self.create_expense_report({
+            'name': 'Expense for John Smith 2',
+            'expense_line_ids': [Command.create({
+                'employee_id': self.expense_employee.id,
+                'product_id': self.product_a.id,
+                'price_unit': 1000.00,
+                'date': '2022-01-25',
             })],
         })
+        expense_sheet_2.action_submit_sheet()
+
+        with freeze_time(datetime(year=2022, month=2, day=25, hour=0, minute=0, second=0)):
```

Comment:
We can also do it that way which is probably more succinct:
```suggestion
        with freeze_time('2022-02-25'):
```